### ISSUE TITLE
[android] Cannot catch WILL_START_LOADING_MAP because I can't attach listener early.

### ISSUE THREAD

**Issue by dapetcu21** [mapbox/mapbox-gl-native#5779]
Map change events happening at initialisation time cannot be captured because `MapView.addOnMapChangedListener` can only be called after the constructor finishes.

In particular, I can't listen to the first `MapView.WILL_START_LOADING_MAP`.

I propose we have a way to specify map change listeners in `MapboxMapOptions`.

Ref: [mapbox/react-native-mapbox-gl#383](https://github.com/mapbox/react-native-mapbox-gl/issues/383)

**Platform:** Android
**Mapbox SDK version:** 4.1.1
### Steps to trigger behavior

``` java
_mapView = new MapView(this.getContext(), _mapOptions);
_mapView.addOnMapChangedListener(this);

@Override
public void onMapChanged(int change) {
    switch (change) {
        case MapView.WILL_START_LOADING_MAP:
            // This line should be reached
            break;
    }
}
```
### Expected behavior

I should be able to know when the map starts loading even if it starts loading immediately.
### Actual behavior

The line isn't reached because the event fires before the event listener is added.

**Comment by tobrun** [comment]
Thank you for the request @dapetcu21

**Comment by tobrun** [comment]
@dapetcu21 I took a quick peek at the `MapboxMapOptions`. This class implements the Parceable interface, which is be used after configuration changes. This requirement makes it an unfit candidate to hold onto the mapchange callback for us (since we won't be able to store/restore that instance). 

Another solution would be to expose a new constructor on MapView that contains 3 params:
 - Context
 - MapboxMapOptions
 - OnMapChangedListener

**Comment by cammace** [comment]
This is fixed in `4.2.0`, you can now capture `WILL_START_LOADING_MAP` before the map is loaded. No need for a new constructor here right @tobrun?
